torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -466,25 +578,271 @@ class CppMicroGemmFP32Vec(CppMicroGemm):
+{%- elif input2_dtype == torch.int8 %}
+                auto a32 = at::vec::convert_to_int32<int8_t>(A + (sM * m + s) * lda + k * VLEN, e_k);
+                va = at::vec::convert<float>(a32);
+{%- else %}
+                va = VectorizedIn::loadu(A + (sM * m + s) * lda + k * VLEN, e_k);
+{%- endif %}
+
+{%- if alpha != 1 %}
+                va = va * Vectorized({{alpha}});
+{%- endif %}
+                if (k == 0) {
+                    {{kernel.unroll_pragma(internal_block_n)}}
+                    for (int i = 0; i < e_n; i++) {
+{%- if input2_dtype in [torch.bfloat16, torch.float16] %}
+                        vmid[sN * s + i] = at::vec::fmadd(va[0], vb[2 * i], Vectorized(0.0f));
+                        vmid[sN * s + i] = at::vec::fmadd(va[1], vb[2 * i + 1], vmid[sN * s + i]);
+{%- else %}
+                        vmid[sN * s + i] = at::vec::fmadd(va, vb[i], Vectorized(0.0f));
+{%- endif %}
+                    }
+                } else {
+                    {{kernel.unroll_pragma(internal_block_n)}}
+                    for (int i = 0; i < e_n; i++) {
+{%- if input2_dtype in [torch.bfloat16, torch.float16] %}
+                        vmid[sN * s + i] = at::vec::fmadd(va[0], vb[2 * i], vmid[sN * s + i]);
+                        vmid[sN * s + i] = at::vec::fmadd(va[1], vb[2 * i + 1], vmid[sN * s + i]);
+{%- else %}
+                        vmid[sN * s + i] = at::vec::fmadd(va, vb[i], vmid[sN * s + i]);
+{%- endif %}
+                    }
+                }
+            }
+
+            // store to C
+            if (k == _K - 1) {
+                {{kernel.unroll_pragma(internal_block_m)}}
+                for (int s = 0; s < e_m; s++) {
+                    {{kernel.unroll_pragma(internal_block_n)}}
+                    for (int i = 0; i < e_n; i++) {
+                        auto v = at::vec::vec_reduce_all([](Vectorized& x, Vectorized& y) { return x + y; }, vmid[sN * s + i]);
+                        if constexpr (accum) {
+                            auto c = *(C + (sM * m + s) * ldc + sN * n + i);
+                            *(C + (sM * m + s) * ldc + sN * n + i) = c + v;
+                        } else {
+                            *(C + (sM * m + s) * ldc + sN * n + i) = v;
+                        }
+                    }
+                }
+            }
+        };
+
+        for (int n = 0; n < bN; ++n) {
+            for (int m = 0; m < bM; ++m) {
+                for (int k = 0; k < _K; ++k) {
+                    compute_trans(m, n, k);
+                }
+            }
+        }
+    }
+{%- endif %}
 }
 """
 
+    # set extra parameters to generate gemm that
+    # supports arbitrary sizes of n and transposed B matrix
+    def set_extra_configs(self, extra_config):
+        if not hasattr(extra_config, "trans_b"):
+            extra_config.trans_b = False
+        if not hasattr(extra_config, "expand_n"):
+            extra_config.expand_n = False
```

Comment:
Can we decide the expand_n based on the size of N and block_n instead of expose the config to user?

Reply:
The setting were removed, leaving only trans_b.